File: benchmark/results/python-invocation-script/run-001/b-baseline/output/metaengine_demo/inventory/services/reservation_service.py
"""ReservationService service."""
from typing import Optional

from metaengine_demo.inventory.aggregates.stock_item import StockItem


class ReservationService:
    """ReservationService service."""

    def create(self, input: Optional[StockItem]) -> StockItem:
        raise NotImplementedError("not implemented")

    def find_by_id(self, id: str) -> Optional[StockItem]:
        raise NotImplementedError("not implemented")

    def list(self, limit: float) -> list[StockItem]:
        raise NotImplementedError("not implemented")

    def delete(self, id: str) -> None:
        raise NotImplementedError("not implemented")

File: benchmark/results/java-invocation-script/run-004/b-baseline/output/com/metaengine/demo/inventory/services/ReservationService.java
package com.metaengine.demo.inventory.services;

import com.metaengine.demo.inventory.aggregates.StockItem;
import java.util.List;

/** ReservationService domain service in the inventory domain. */
public class ReservationService {
    /** Reserve a StockItem from a partial input. */
    public StockItem create(StockItem input) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Find a StockItem by id, or null if not found. */
    public StockItem findById(String id) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** List StockItems up to limit. */
    public List<StockItem> list(int limit) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Delete a StockItem reservation by id. */
    public void delete(String id) {
        throw new UnsupportedOperationException("not implemented");
    }
}
